test/Interop/Cxx/value-witness-table/custom-destructors.swift

```diff
@@ -0,0 +1,140 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-cxx-interop)
+//
+// REQUIRES: executable_test
+
+import CustomDestructorTypes
+import StdlibUnittest
+
+var CXXDestructorTestSuite = TestSuite("CXXDestructor")
+
+protocol InitWithPtr {
+  init(value: UnsafeMutablePointer<Int32>!)
+}
+
+extension HasUserProvidedDestructor : InitWithPtr { }
+
+protocol InitWithMember {
+  init(member: HasUserProvidedDestructor)
+}
+
+extension HasEmptyDestructorAndMemberWithUserDefinedConstructor
+  : InitWithMember { }
+
+// These are all maked as "@inline(never)" so that the destructor is invoked in
+// this function and not in the caller (after being inlined).
+@inline(never)
+func createTypeWithUserProvidedDestructor(_ ptr: UnsafeMutablePointer<Int32>) {
+  var obj = HasUserProvidedDestructor(value: ptr)
```

Comment:
More idiomatic. (Here and elsewhere in the PR.)
```suggestion
  _ = HasUserProvidedDestructor(value: ptr)
```